tv_apps/smart-home/js/home.js

```diff
@@ -130,9 +157,71 @@
       });
     },
 
-    onCardInserted: function(card, idx) {
-      this.cardScrollable.insertNodeBefore(this._createCardNode(card), idx);
+    _onPinInit: function(newCardButtonElem) {
+      newCardButtonElem.classList.add('new-card');
+      newCardButtonElem.classList.add('new-card-transition');
+
+      // add sliding transition to card-list and every existing cards
+      var i;
+      var cardElem;
+      for(i = 0; i < this.cardListElem.children.length; i++) {
+        cardElem = this.cardListElem.children[i];
+        cardElem.classList.add('card-slide');
+      }
+      this.cardListElem.classList.add('card-slide');
+    },
+
+    _onCardListSlideEnd: function(newCardButtonElem, idx) {
+      this.cardScrollable.setOtherNodesPosition(idx);
```

Comment:
As discussed earlier, may you try to move most of the logics into `scrollable.js`. If there are some difficulty feel free to ping me.